### PR TITLE
Add slam tool to consensus server repo

### PR DESCRIPTION
**Pull request by garbageslam**
The slam tool is a load testing tool for the consensus servers.
Generally, test code should be versioned with the code under test.

This moves the slam tool to the same repo as the consensus servers,
and will make it easier to run this test in CD and more easily use
it to validate changes to the consensus servers.

### CODE REVIEW

File: slam/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[package]
+name = "slam"
+version = "0.13.0"
+authors = ["drakeley <sara.drakeley@gmail.com>"]
```
Comment (eranrund) [inline slam/Cargo.toml:4]: I believe we just put `MobileCoin` here.

File: slam/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+In order to create transactions, Slam requires a copy of the ledger and the set of private keys that own the contents of the ledger. The ledger and keys must **exactly** match those used by the consensus network; if not, Slam's transactions will likely be rejected with `InvalidTxOutMembershipProof` or `LedgerDbError`.
+
+## Generating a local ledger and keys
+If you know how the consensus network's ledger was initialized, you can initialize the same ledger locally with:
+
+```
+    internal $ mkdir -p target/sample_data
+    internal $ cd sample_data
+    internal/target/sample_data $ cargo run --manifest-path=../../public/Cargo.toml -p mc-util-keyfile \
```
Comment (eranrund) [inline slam/README.md:21]: This needs to be updated

File: slam/src/config.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
+//! Configuration parameters for the slam script
```
Comment (eranrund) [inline slam/src/config.rs:1]: This needs a copyright notice

**Review by eranrund — CHANGES_REQUESTED** [review]
Some documentation things need to be improved but we know the code is doing its job :)

**Comment by garbageslam** [comment]
@jcape lmk if the updates for `HardcodedCredentialsProvider` looks right, i mostly copied from `mobilecoind`

Do you guys think the crate should be `mc-slam` now?

**Comment by jcape** [comment]
> Do you guys think the crate should be `mc-slam` now?

Yep (will review inline for the HCP stuff)

**Review by jcape — APPROVED** [review]
LGTM, assuming it runs properly against deployed networks. I'd remove some of the weird stuff from `README.md`, though, like references to specific networks or repositories that people may not have access to...

**Comment by garbageslam** [comment]
TODO:
-[] deploy and run slam as in instructions

**Review by sugargoat — APPROVED** [review]

**Pull request merged**